docker/build-tools/Dockerfile

```diff
@@ -92,13 +92,22 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     xz-utils
 
 # Install protoc
-ADD https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip /tmp/
-RUN unzip /tmp/protoc-${PROTOC_VERSION}-linux-x86_64.zip
-RUN mv /tmp/bin/protoc ${OUTDIR}/usr/bin
+RUN set -eux; \
+    \
+    case $(uname -m) in \
+        x86_64) export PROTOC_ZIP=protoc-${PROTOC_VERSION}-linux-x86_64.zip;; \
+        aarch64) export PROTOC_ZIP=protoc-${PROTOC_VERSION}-linux-aarch_64.zip;; \
+        *) echo "unsupported architecture"; exit 1 ;; \
+    esac; \
+    \
+    wget -O /tmp/${PROTOC_ZIP} https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/${PROTOC_ZIP}; \
+    unzip /tmp/${PROTOC_ZIP}; \
+    mv /tmp/bin/protoc ${OUTDIR}/usr/bin
 
 # Install gh
-ADD https://github.com/cli/cli/releases/download/v${GH_VERSION}/gh_${GH_VERSION}_linux_amd64.deb /tmp/
-RUN dpkg -i /tmp/gh_${GH_VERSION}_linux_amd64.deb
+ARG TARGETARCH
```

Comment:
Can we use default TARGETARCH=amd64?

Reply:
if we use `docker buildx build`, this arg will pass by `buildx`

https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

if we want to support `docker build` too, need to use `${TARGETARCH:-amd64}`